src/components/side_panel/side_panel/side_panel.ts

```diff
@@ -152,37 +154,25 @@ css/* scss */ `
   }
 `;
 
-interface Props {
-  component: string;
-  panelProps: any;
-  onCloseSidePanel: () => void;
-}
-
-interface State {
-  panel: SidePanelContent;
-}
-
-export class SidePanel extends Component<Props, SpreadsheetChildEnv> {
+export class SidePanel extends Component<never, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-SidePanel";
-  static props = {
-    component: String,
-    panelProps: { type: Object, optional: true },
-    onCloseSidePanel: Function,
-  };
-  state!: State;
+  static props = {};
+  sidePanelStore!: Store<SidePanelStore>;
 
   setup() {
-    this.state = useState({
-      panel: sidePanelRegistry.get(this.props.component),
-    });
-    onWillUpdateProps(
-      (nextProps: Props) => (this.state.panel = sidePanelRegistry.get(nextProps.component))
-    );
+    this.sidePanelStore = useStore(SidePanelStore);
+  }
+
+  get panel() {
+    return sidePanelRegistry.get(this.sidePanelStore.component);
```

Comment:
Make the store expose directly the component maybe ? No reason for the SidePanelComponent to worry about a registry if we want to make it dumber

Reply:
But the `sidePanelStore.component` as a string is still needed in the `t-key` of the templates ... Are the `t-key` still necessary ? it should be reactive now, no ?

Reply:
I agree in principle. But it creates a circular dependency. We'd have to re-organize quite a bit of things